// crow-scripts/configs/rollup.js
const { rollup } = require('rollup')
const babel = require('rollup-plugin-babel')
const resolve = require('rollup-plugin-node-resolve')
const commonjs = require('rollup-plugin-commonjs')
const crowPreset = require('babel-preset-crow')
const path = require('path')

module.exports = ({
  entry,
  useBuiltIns,
  loose,
  spec,
  debug,
  targets,
}) => ({
  entry,
  external: (id) => {
    const ext = path.extname(id)
    // files without extension are passed through
    if (ext === '') return false
    // files with node extensions are passed through
    if (/\.js(x|on)?/.test(ext)) return false
    // file with extensions like css and images are marked as external
    return true
  },
  plugins: [
    resolve({
      jail: path.resolve(process.cwd(), path.dirname(entry)),
    }),
    commonjs(),
    babel({
      presets: [
        crowPreset({
          modules: false,
          useBuiltIns,
          loose,
          spec,
          debug,
          targets,
          extendPlugins: [
            require('babel-plugin-external-helpers')
          ]
        })
      ]
    }),
  ]
})
