// crow-scripts/configs/webpack/base-config.js
const path = require('path')
const webpack = require('webpack')

module.exports = ({
  entry,
  libraryName,
  libraryTarget = "commonjs2",
  environment = 'production',
} = {}) => (config) =>  {
  const baseConfig = {
    target: 'web',
    devtool: 'source-map',
    entry,
    output: {
      path: path.resolve(process.cwd()),
      filename: '[name].[chunkhash:8].bundle.js',
      chunkFilename: '[name].[chunkhash:8].chunk.js',
      library: libraryName,
      libraryTarget,
    },
    module: {
      rules: [
        {
          oneOf: []
        }
      ]
    },
    plugins: [
      new webpack.EnvironmentPlugin({
        NODE_ENV: environment,
      }),
      // https://github.com/jmblog/how-to-optimize-momentjs-with-webpack
      new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/)
    ],
    // https://github.com/facebookincubator/create-react-app/blob/master/packages/react-scripts/config/webpack.config.prod.js
    // Some libraries import Node modules but don't use them in the browser.
    // Tell Webpack to provide empty mocks for them so importing them works.
    node: {
      dgram: 'empty',
      fs: 'empty',
      net: 'empty',
      tls: 'empty',
    },
  }

  return(Object.assign({}, config, baseConfig))
}
